### PR TITLE
new_test/5.1/test_requires_unified_shared_memory_heap_is_device_ptr.F90

### PR DESCRIPTION
**Pull request by seyonglee**
Add new_test/5.1/test_requires_unified_shared_memory_heap_is_device_ptr.F90
- XL 16.1.1-10: Fortran test failed.
- NVHPC 22.5: Fortran test failed.
- GCC 12.1.1: Fortran test failed.

### CODE REVIEW

File: tests/5.1/requires/test_requires_unified_shared_memory_heap_is_device_ptr.F90 [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      anArray(i) = i
+      anArrayCopy(i) = 0
+    END DO
+
+    ! Modify in the device
+    !$omp target is_device_ptr(aPtr)
+    DO i = 1, N
+      aPtr(i) = aPtr(i) + 10
+    END DO
+    !$omp end target
```
Comment (tob2) [inline tests/5.1/requires/test_requires_unified_shared_memory_heap_is_device_ptr.F90:66]: `aPtr` is `type(c_ptr)` – and can not be accessed as array. How about:

```suggestion
    !$omp target is_device_ptr(aPtr)
    BLOCK
      integer, pointer :: anPtr(:)
      call c_f_pointer(aPtr, anPtr, [N])
      DO i = 1, N
        anPtr(i) = anPtr(i) + 10
      END DO
    END BLOCK  ! note: no '!$omp end target' as this is a 'strictly structured block'
```
which declares a local pointer variable.

_Regarding the `is_device_ptr` failure in GCC, that's a missing OpenMP 5.1 feature, tracked at https://gcc.gnu.org/PR105318_

Comment (seyonglee) [reply tests/5.1/requires/test_requires_unified_shared_memory_heap_is_device_ptr.F90]: Thank you for catching this!

**Comment by seyonglee** [comment]
This is OpenMP V5.1 version (use C_PTR variable for the `is_device_ptr` clause) of #585 (use dummy argument for the `is_device_ptr` clause).

**Review by spophale — APPROVED** [review]

**Comment by spophale** [comment]
@mjcarr458 will review.

**Review by mjcarr458 — APPROVED** [review]
Looks good

**Comment by seyonglee** [comment]
[Update on Nov. 27, 2022]
        - OpenMP V5.1 version of PR #585
        - XL 16.1.1-10: Fortran test failed.
        - NVHPC 22.5: Fortran test failed (NVFORTRAN-S-1219-Unimplemented feature: BLOCK construct in the scope of a parallel directive).
        - *GCC 12.2.1: Fortran test failed (Error: Non-dummy object 'aptr' in IS_DEVICE_PTR clause).

**Pull request merged**